lib/sisimai/bite/email/gsuite.rb

```diff
@@ -0,0 +1,273 @@
+module Sisimai::Bite::Email
+  module GSuite
+    # Sisimai::Bite::Email::GSuite parses a bounce email which created by G Suite.
+    # Methods in the module are called from only Sisimai::Message.
+    class << self
+      # Imported from p5-Sisimail/lib/Sisimai/Bite/Email/GSuite.pm
+      require 'sisimai/bite/email'
+
+      Re0 = {
+        :from    => %r/[@]googlemail[.]com[>]?\z/,
+        :subject => %r/Delivery[ ]Status[ ]Notification/,
+      }
+      Re1 = {
+        :begin   => %r/\A[*][*][ ].+[ ][*][*]\z/,
+        :error   => %r/\AThe[ ]response([ ]from[ ]the[ ]remote[ ]server)?[ ]was:\z/,
+        :html    => %r{\AContent-Type:[ ]*text/html;[ ]*charset=['"]?(?:UTF|utf)[-]8['"]?\z},
+        :rfc822  => %r{\AContent-Type:[ ]*(?:message/rfc822|text/rfc822-headers)\z},
+        :endof   => %r/\A__END_OF_EMAIL_MESSAGE__\z/,
+      }
+      ErrorMayBe = {
+        :userunknown  => %r/because the address couldn't be found/,
+        :notaccept    => %r/Null MX/,
+        :networkerror => %r/DNS type .+ lookup of .+ responded with code NXDOMAIN/,
+      }
+      Indicators = Sisimai::Bite::Email.INDICATORS
+
+      def description; return 'G Suite: https://gsuite.google.com'; end
+      def smtpagent;   return Sisimai::Bite.smtpagent(self); end
+      def headerlist;  return ['X-Gm-Message-State']; end
+      def pattern;     return Re0; end
+
+      # Parse bounce messages from G Suite (Transfer from G Suite to a destinaion host)
+      # @param         [Hash] mhead       Message headers of a bounce email
+      # @options mhead [String] from      From header
+      # @options mhead [String] date      Date header
+      # @options mhead [String] subject   Subject header
+      # @options mhead [Array]  received  Received headers
+      # @options mhead [String] others    Other required headers
+      # @param         [String] mbody     Message body of a bounce email
+      # @return        [Hash, Nil]        Bounce data list and message/rfc822
+      #                                   part or nil if it failed to parse or
+      #                                   the arguments are missing
+      def scan(mhead, mbody)
+        return nil unless mhead
+        return nil unless mbody
+
+        return nil unless mhead['from']    =~ Re0[:from]
+        return nil unless mhead['subject'] =~ Re0[:subject]
+        return nil unless mhead['x-gm-message-state']
+
+        require 'sisimai/address'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        hasdivided = mbody.split("\n")
+        rfc822list = []     # (Array) Each line in message/rfc822 part string
```

Comment:
__[rubocop]__
* Metrics/AbcSize: Assignment Branch Condition size for scan is too high. [162.4/30] [:ref](http://c2.com/cgi/wiki?AbcMetric)

<!-- Comment by [SideCI](https://sideci.com) -->